File: cpp/vm/advm/token.cpp
#include "token.h"

Token::Token() {
    type = TT_UNDEFINED;
};

Token::Token(TokenType t) {
    type = t;
}

Token::Token(TokenType t, std::string l) {
    type = t;
    stringLiteral = l;
}

std::string Token::toString() {
    return "Token [" + tokenTypeConverter[type] + "] <" + stringLiteral + ">";
}

File: java/Basic2/src/main/java/token/Token.java
package token;

public class Token {
    TokenTypeEnum type;
    String tokenLiteral;

    public Token() {

    }

    public Token(TokenTypeEnum type, String tokenLiteral) {
        this.type = type;
        this.tokenLiteral = tokenLiteral;
    }

    public TokenTypeEnum getType() {
        return type;
    }

    public void setType(TokenTypeEnum type) {
        this.type = type;
    }

    public String getTokenLiteral() {
        return tokenLiteral;
    }

    public void setTokenLiteral(String tokenLiteral) {
        this.tokenLiteral = tokenLiteral;
    }

    public String toString() {
        return "Token [" + TokenTypeConverter.covertToken(getType()) + "] <" + getTokenLiteral() + ">";
    }
}
